torch/distributed/distributed_c10d.py

```diff
@@ -379,7 +337,10 @@ def init_process_group(backend,
         assert world_size > 0, 'world_size must be positive if using store'
         assert rank >= 0, 'rank must be non-negative if using store'
     elif init_method is None:
-        init_method = "env://"
+        if is_mpi_available() and "OMPI_COMM_WORLD_SIZE" in os.environ:
```

Comment:
This env variable seems very OpenMPI specific, while people might be using e.g. MPICH or the Intel implementations